addons/web/static/src/search/action_menus/action_menus.js

```diff
@@ -52,21 +54,14 @@ export class ActionMenus extends Component {
         this.actionService = useService("action");
         onWillStart(async () => {
             this.actionItems = await this.getActionItems(this.props);
+            this.printItems = await this.getPrintItems();
         });
         onWillUpdateProps(async (nextProps) => {
```

Comment:
Hooking on onWillUpdateProps is often a bad idea. Especially here as you do rpcs, i.e. as you really don't want to do the work when it's not necessary. onWillUpdateProps could be called even if the selection didn't change, and the selection could change without onWillUpdateProps being called. I guess that you would like to hook on the moment where the selection changes. Also, as Rémy suggested, the rpcs should be done when we the user toggles the dropdown, not before (useless most of the cases).

Reply:
You should probably use the beforeOpen props of dropdowns, which allows to execute an async callback before opening the dropdown